### ISSUE TITLE
Code to be modified to handle new columns in plugin excel sheets

### ISSUE THREAD

**Issue by rajeshgrs** [rajeshgrs/dq-cli#8]
Attached is the new plugin excel sheet "oie.xlsx"
    a) "Column Name" in the first column should be considered as the plugin name to fetch for corresponding Templates. Currently we use to fetch by worksheet name.
        This new column is added to get the TestCaseID for each command we run. 
    b) "TestCaseID" column in the excel sheet should be used as log_file name in the templates. So we know that, for this particular test_case we should refer to log file that matches with TetsCaseID column.

[oie.xlsx](https://github.com/rajeshgrs/dq-cli/files/693258/oie.xlsx)

**Comment by rajeshgrs** [comment]
Priority 1: https://github.com/rajeshgrs/dq-cli/issues/8
              https://github.com/rajeshgrs/dq-cli/issues/9

Priority 2: https://github.com/rajeshgrs/dq-cli/issues/11

Priority 3: https://github.com/rajeshgrs/dq-cli/issues/10

**Comment by jeevanjagadeesh** [comment]
@rajeshgrs , I understand the use of TestCaseID column. 
But dont see much use of "Plugin Name" in first column, also the values in the first column are Run , skip which is already taken care in "ExecuteOption" column. 
Can we still use the worksheet name has plugin name?

**Comment by jeevanjagadeesh** [comment]
@rajeshgrs , if we have same name between the column name and template replace param, then no code change is required.

i.e  column name is "TestCaseID", so if we have <<TestCaseID>> in the template instead of <<LogFileName>>. Then the current code will take care of replacing the with dynamic value

**Comment by rajeshgrs** [comment]
Ok Jeevan, Can we do like this.

 For now lets assume, instead of keeping the column name as
"ExecutionOption", we are naming it with "{plugin_name}". I will describe
the usefulness of this column if we make the code act in below manner:-

Can we make the code to traverse in this manner ->
       step-1) Code will get into workbook
       step-2) Traverse through every worksheet in a loop
                    2.a) It finds whether any executable "Run" action given
in first column
                    2.b) If "Run" Found, then it will parse the data in 2nd
column i.e, "TestID" which will have name of plugin before underscore
character. So based on this plugin name, code will loopup the template and
replace the values using input from 3rd column onwards.
                    2.c) Code will traverse through all the rows in excel
irrespective of whether it has data or not (This is to allow entering of
more than one plugin data in a single worksheet, instead having too many
tabs in the workbook)

Let me know what you think.


Thanks,
Rajesh

**Comment by rajeshgrs** [comment]
Yes, the TestCaseID is more useful to track the status of command
execution. If you see other list of issues I have raised, I have
specifically mentioned about this one. Or else is there a better
alternative way to track the status of each command execution? At end
point, if any failure occurs, I should be able to quickly get to the right
place in the input excel sheet without any hassles.

**Comment by rajeshgrs** [comment]
Attached "QATrackerReport.xml" is the XML file format that we should be
targeting to get the status of each command execution.
